### ISSUE TITLE
Option to rebuild the cache in the background

### ISSUE THREAD

**Issue by vermiculus** [bbatsov/projectile#1515]
Howdy! As I'm sure more than a few Projectile users do, I work with _very_ large repositories. So large that reconstructing the cache takes ...(runs `time svn list -R`)... 21 minutes. New files I need to access are being added regularly, so I'm finding that I need to rebuild the cache far more often than I used to.

I can do the development myself if desired, but I wanted to get your thoughts: is there any reason we can't run `projectile-svn-command` (or anything from `projectile-indexing-method => 'alien`) as a background process and consume the output when the command completes? Do you foresee any challenges?

**Comment by bbatsov** [comment]
No, this shouldn't be particularly hard to achieve. The only thing we'll have to be mindful is would be not messing up with the existing cache while the rebuild operation is in progress (probably it should write some to some tmp cache that replaces that one in use). Generally your suggestions goes well with some plans I have to remove the big cache file and replace it with one cache file per project. Unfortunately I never got to working on this.

**Comment by goldfita** [comment]
I was thinking of something similar. With cache disabled, it takes several seconds for search for files to come up. If the indexing could happen in the background while I started typing, I might barely even notice it was happening.

**Comment by bbatsov** [comment]
There's `projectile-index-project-async` now, which warms the project cache in a background process without freezing Emacs, plus optional async indexing so even a cold `projectile-find-file` doesn't block. That's exactly the "rebuild the cache in the background" ask.

Closing as done.
